content/shared/tutorials/_testing_push_android.textile

```diff
@@ -0,0 +1,101 @@
+Now that our app is ready, lets test it out. Before using any of these methods make sure you have built and run the app on a real Android device and accepted push notifications.
+
+h3. a. Testing using Device ID
+
+Add the following method to your index.js and replace "YOUR_DEVICE_ID" with the id you just retrieved from your console:
+
+```[nodejs]
+app.get('/push/device', function (req, res) {
+  var recipient = {
+    deviceId: 'YOUR_DEVICE_ID'
+  };
+  var notification = {
+    notification: {
+      title: 'Hello from Ably!'
+    }
+  };
+  realtime.push.publish(recipient, notification, function(err) {
+    if (err) {
+      console.log('Unable to publish push notification; err = ' + err.message);
+      return;
+    }
+    console.log('Push notification published');
+    res.send("Push Sent");
+  });
+})
+```
+
+Restart the server and navigate to @YOUR-SERVER-URL/push/device@ in your browser. You should be able to receive a notification on your device as shown the image below.
+
+<a href="/images/tutorials/android-push/android-push-screenshot.jpg" target="_blank">
+	 <img src="/images/tutorials/android-push/android-push-screenshot.jpg" style="width: 50%; height: 50%" alt="Android Push Notification Screenshot">
+</a>
+
+h3. b. Testing using Client ID
+
+We can test sending push notifications using the client ID, this allows us to push all devices with that client ID by adding the following code to your index.js file:
```

Comment:
I think there should be a full stop between `client ID` and `this allows...`, feels a bit stilted as is to me. Same boldening of `index.js` point.